### ISSUE TITLE
MemberSaved Notification gives wrong key for entity

### ISSUE THREAD

**Issue by warrenbuckley** [umbraco/Umbraco-CMS#15968]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

13.2.2

### Bug summary

When using the `MemberSaved` notification the entity with the notification has a different GUID/Key for the newly created member and is different compared to `IMemberManager` and `IMemberService`

When reproducing this I noticed that the event/notification for MemberSaved is executed 3 times for a new member that is created.

### Specifics

I have checked this in isolation away from the client project to ensure its not a project specific bug and was able to reproduce this in a clean version of 13.2.2

I have attached a video to help show the problem 👀 

https://github.com/umbraco/Umbraco-CMS/assets/1389894/92be1c22-2762-48fc-82c4-26c2f3aa1458

### Steps to reproduce

Use the sample code below and step through with a debugger to see the variables in this sample code are different when you create a brand new member that has a single member group assigned to them.

```csharp
using Umbraco.Cms.Core.Composing;
using Umbraco.Cms.Core.Events;
using Umbraco.Cms.Core.Models.Entities;
using Umbraco.Cms.Core.Notifications;
using Umbraco.Cms.Core.Security;
using Umbraco.Cms.Core.Services;

namespace MemberNotificationBug
{
	public class AddMemberNotification : IComposer
	{
		public void Compose(IUmbracoBuilder builder)
		{
			builder.AddNotificationAsyncHandler<MemberSavedNotification, MemberIsSavedNotification>();
		}
	}

	public class MemberIsSavedNotification : INotificationAsyncHandler<MemberSavedNotification>
	{
		private IMemberManager _memberManager;
		private IMemberService _memberService;
		private ILogger<MemberIsSavedNotification> _logger;

		public MemberIsSavedNotification(IMemberManager memberManager, IMemberService memberService, ILogger<MemberIsSavedNotification> logger)
        {
			_memberManager = memberManager;
			_memberService = memberService;
			_logger = logger;
		}

		public async Task HandleAsync(MemberSavedNotification notification, CancellationToken cancellationToken)
		{
			foreach (var memberEntity in notification.SavedEntities)
			{
				var dirty = (IRememberBeingDirty)memberEntity;
				var isNewMember = dirty.WasPropertyDirty("Id");
				var memberChangedToApproved = dirty.WasPropertyDirty("IsApproved");


				var memberFromManager = await _memberManager.FindByNameAsync(memberEntity.Username);
				if (memberFromManager == null)
				{
					throw new ArgumentNullException("No member?!");
				}

				// BUG: Why is the key different from the entity in the notification from one via memberManager ?
				var memberFromNotificationKey = memberEntity.Key;
				var memberFromManagerKey = memberFromManager.Key;

				var memberFromService = _memberService.GetByUsername(memberEntity.Username);
				var memberFromServiceKey = memberFromService?.Key;

				_logger.LogInformation("Notification: {memberFromNotificationKey} Manager: {memberFromManagerKey} Service: {memberFromServiceKey}", memberFromNotificationKey, memberFromManagerKey, memberFromServiceKey);

			}
		}			
	}
}
```

### Expected result / actual result

I expect the GUID/Key of the entity with the notification to be the correct one and not differ when compared to `IMemberManager` and `IMemberService`

**Comment by Zeegaan** [comment]
I can definitely reproduce this 😁
I had a look into this code, just cause I was curious of what is triggering multiple notifications.

It seems that when we call `IdentityResult created = await _memberManager.CreateAsync(identityMember, contentItem.Password?.NewPassword!);`
It uses the member store (as you would expect).
But then the member store underneath calls the memberService.Save() here:

![image](https://github.com/umbraco/Umbraco-CMS/assets/70372949/85a0ad0f-a450-4f26-8b65-7d09ee7428a8)
This is where we have the "wrong" key, cause it gets updated a little bit later in the `MemberController`:

![image](https://github.com/umbraco/Umbraco-CMS/assets/70372949/10711616-f4a6-4a19-9f11-472355ac1cb0)

And then we call save again, 17 lines later:

![image](https://github.com/umbraco/Umbraco-CMS/assets/70372949/3a2dbc80-0443-4793-92f7-5aece20496f6)

So in summary, what's happening right now is:
- Create member
-Update member properties with all the content properties
- Save member again

**Comment by Zeegaan** [comment]
While I think saving the member twice is fundamentally wrong, it's breaking to fix it atm, (might be rework worthy for V14/V15)

I think what would fix this bug for now, is to pass the actual given get when we create the member identy user:
![image](https://github.com/umbraco/Umbraco-CMS/assets/70372949/403a5e73-caa9-4112-bae9-9c7aeecc8aef)
So instead of the above, i would implement a new `CreateNew()` method, that also takes the key as a parameter, and sets it 👍 

I will mark this up for grabs, as there is hopefully enough to go in, with the info both you and I have provided here 🙈

**Comment by warrenbuckley** [comment]
Thanks @Zeegaan for this, good to hear you are reliably able to reproduce this.
Just that I don't have time right now, to do a PR or fix for this, can you ensure this does not get closed off because I don't have the time & this gets forgotten.

**Comment by biapar** [comment]
Which I s the solution?

**Comment by warrenbuckley** [comment]
Friendly bump as could be ideal for v15 or v16 perhaps @Zeegaan ?

**Comment by biapar** [comment]
I solved to chance the propierties into the saving handle and not into the saved.

**Comment by andersreus** [comment]
Hey @warrenbuckley 
I reproduced the issue on 13.7.0 - where the member notification key differs from the IMemberManager and IMemberService keys.

I also tested this on the latest 13/dev with the code sample you provided and all three keys are identical. 
So I assume the issue is fixed and will be part of 13.8.0 🙂
